src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,454 @@
+        )
+
+    @property
+    def _client(self) -> kraken:
+        super_client: Exchange = super()._client
+        if not isinstance(super_client, kraken):
+            raise RP2RuntimeError("Exchange is not instance of class kraken.")
+        return super_client
+
+    def _get_process_deposits_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _get_process_withdrawals_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _get_process_trades_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        pass
+
+    def _process_gains(
+        self,
+        in_transactions: List[InTransaction],
+        out_transactions: List[OutTransaction],
+    ) -> None:
+        pass
+
+    def _gather_api_data(self) -> Tuple[Any, Any]:
+        loaded_cache = load_from_cache(self.__CACHE_FILE)
+        if self.use_cache and loaded_cache:
+            return loaded_cache
+
+        # get initial trade history to get count
+        index: int = 0
+        count: int = int(self._client.private_post_tradeshistory(params={_OFFSET: index})[_RESULT][_COUNT])
+        trade_history: Dict[str, Dict[str, Union[str, int, None, List[str]]]] = {}
+        while index < count:
+            trade_history.update(self._process_trade_history(index))
+            index += _TRADE_RECORD_LIMIT
+
+        # reset index and count for next API call
+        index = 0
+        count = int(self._client.private_post_ledgers(params={_OFFSET: index})[_RESULT][_COUNT])
+        ledger: Dict[str, Dict[str, Union[str, int, None, List[str]]]] = {}
+        while index < count:
+            ledger.update(self._process_ledger(index))
+            index += _TRADE_RECORD_LIMIT
+
+        result = (trade_history, ledger)
+
+        if self.use_cache:
+            save_to_cache(self.__CACHE_FILE, result)
+
+        return result
+
+    def load(self, country: AbstractCountry = us.US()) -> List[AbstractTransaction]:
+        (trade_history, ledger) = self._gather_api_data()
+        return self._compute_transaction_set(trade_history, ledger)
+
+    def _compute_transaction_set(self, trade_history: Dict[str, Dict[str, Union[str, int, None, List[str]]]], ledger: Dict[str, Dict[str, Union[str, int, None, List[str]]]]) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        unhandled_types: Dict[str, str] = {}
+        for key in ledger:
+            record: Dict[str, str] = ledger[key]
+            self.__logger.debug("Ledger record: %s", record)
+
+            timestamp_value: str = self._rp2_timestamp_from_seconds_epoch(record[_TIMESTAMP])
+
+            is_fiat_asset: bool = record[_ASSET] in _KRAKEN_FIAT_LIST
+
+            amount: RP2Decimal = abs(RP2Decimal(record[_AMOUNT]))
+            asset_base: str = self.base_id_to_base[record[_ASSET]]
+            raw_data = str(record)
+
+            if record[_TYPE] == _WITHDRAWAL or record[_TYPE] == _DEPOSIT:
```

Comment:
Nit. It's bit more Pythonic this way:
```suggestion
            if record[_TYPE] in {_WITHDRAWAL, _DEPOSIT}:
```

Reply:
That's fine. I like that notation a lot I just don't always remember to use it. One thing, I think sets don't work for boolean statements and it has to be list(). I don't know why.

Reply:
It works with sets: see https://github.com/eprbell/dali-rp2/blob/main/src/dali/plugin/input/csv/trezor.py#L94